api/v1beta1/azuremanagedcontrolplane_default.go

```diff
@@ -90,11 +87,8 @@ func (m *AzureManagedControlPlane) setDefaultSubnet() {
 	if m.Spec.VirtualNetwork.Subnet.Name == "" {
 		m.Spec.VirtualNetwork.Subnet.Name = m.Name
 	}
-	if m.Spec.VirtualNetwork.Subnet.CIDRBlock == "" {
```

Comment:
> For pull-cluster-api-provider-azure-e2e-aks
> 
> ```
> Expected success, but got an error:
>     <errors.aggregate | len:1, cap:1>: 
>     admission webhook "validation.azuremanagedcontrolplanes.infrastructure.cluster.x-k8s.io" denied the request: [spec.virtualNetwork.CIDRBlock: Invalid value: "": pre-existing virtual networks CIDR block is invalid, spec.virtualNetwork.Subnet.CIDRBlock: Invalid value: "": pre-existing subnets CIDR block is invalid]
>     [
> ```
> 
> Not sure how these would be empty when they are defaulted [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/33a15a6fda3fa23c884a592826aeb87baefa7db5/api/v1beta1/types_class.go#L425-L424) and [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/33a15a6fda3fa23c884a592826aeb87baefa7db5/api/v1beta1/azuremanagedcontrolplane_types.go#L390).
> 
> Does the e2e use any API changes from the PR?

Webhooks are 100% invoked by the e2e tests. I'm not exactly sure why this specific error is cropping up, but the logic around here seems non-trivial so I'd be okay to omit the CIDR block changes from this PR. My main idea with #4204 was to pick off the low-hanging fruit.

Reply:
Ok sounds good. I'll put those back.